x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -291,6 +295,12 @@ export class FleetPlugin
     licenseService.start(this.licensing$);
     agentCheckinState.start();
 
+    const fleetServerEnabled = appContextService.getConfig()?.agents?.fleetServerEnabled;
+    if (fleetServerEnabled) {
+      await this.licensing$.pipe(first()).toPromise();
```

Comment:
why do we wait for licensing before starting migration?

Reply:
we need licensing to be available so the SO client is correctly initialized.